test/k8sT/Services.go

```diff
@@ -432,6 +432,84 @@ var _ = Describe("K8sServicesTest", func() {
 		})
 	})
 
+	SkipContextIf(helpers.RunsWithKubeProxy, "Checks connectivity when skipping socket lb in pod ns", func() {
+		var (
+			demoDSYAML string
+			demoYAML   string
+		)
+
+		BeforeAll(func() {
+			DeployCiliumOptionsAndDNS(kubectl, ciliumFilename, map[string]string{
+				"skipSocketLBInPodNamespace": "true",
+			})
+			demoDSYAML = helpers.ManifestGet(kubectl.BasePath(), "demo_ds.yaml")
+			res := kubectl.ApplyDefault(demoDSYAML)
+			res.ExpectSuccess("Unable to apply %s", demoDSYAML)
+			demoYAML = helpers.ManifestGet(kubectl.BasePath(), "demo.yaml")
+			res = kubectl.ApplyDefault(demoYAML)
+			res.ExpectSuccess("unable to apply %s", demoYAML)
+			waitPodsDs()
+			err := kubectl.WaitforPods(helpers.DefaultNamespace, "-l zgroup=testapp", helpers.HelperTimeout)
+			Expect(err).Should(BeNil())
+		})
+
+		AfterAll(func() {
+			_ = kubectl.Delete(demoDSYAML)
+			_ = kubectl.Delete(demoYAML)
+			ExpectAllPodsTerminated(kubectl)
+		})
+
+		It("Checks ClusterIP connectivity on the same node", func() {
```

Comment:
Maybe we can add a check for host netns translations (probably assertion will be based on `cilium monitor` output)?

Reply:
host netns translation is implied by the connectivity from host to service, no? I mean if that translation does't happen then host ns should not be able to talk to service backend.

Reply:
Yeah, unless it runs with kube-proxy.